### PR TITLE
feat: add kotlin support

### PR DESCRIPTION
**Pull request by dgschwindturo**
In general, we need to add support for Kotlin via sdkman. One additional requirement is that sdkman cannot install kotlin if java is not already installed, hence why we now have a tier 1 and a tier 2 registry.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,23 +16,41 @@ export default async function run() {
...
+        for (let tier1Tool of Tool.allTier1()) {
             try {
                 // For some reason this catch call isn't working the way I expect,
                 // but I can't figure it out, so we double down with try/catch
-                await tool.setup(core.getInput(tool.name)).catch(errHandler)
+                await tier1Tool
+                    .setup(core.getInput(tier1Tool.name))
+                    .catch(errHandler)
+            } catch (err) {
+                errHandler(err)
```
Comment (tagoro9) [inline index.js:46]: If a tier1 tool fails, should we even try to install tier2 tools? Assuming that kotlin depends on java, it means that kotlin will probably fail too

Comment (tagoro9) [reply index.js:46]: I don't know if you want to do tiers or better have tool dependencies and use a different model

File: kotlin.js [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+import path from "path"
+import assert from "assert"
+import process from "process"
+import fsPromises from "fs/promises"
+
+import core from "@actions/core"
+
+import Tool from "./tool.js"
+
+export default class Kotlin extends Tool {
```
Comment (tagoro9) [inline kotlin.js:11]: It seems this tool and java are very similar (as they are both managed via sdman). Maybe it's worth having an abstract class or delegate class that does all the sdkman stuff and the kotlin and java tools just invoke it

**Review by tagoro9 — COMMENTED** [review]
Took a super quick look. Added some thoughts about tiers vs explicit tool dependencies and also consolidating the kotlin and java tools shared logic into an an sdkman installer

File: index.js (hunk 1 above)
Comment (dgschwindturo) [reply index.js:46]: I suggested having a Tool depends on Tool kind of relationship, @shakefu suggested tiering.

File: kotlin.js (hunk 2 above)
Comment (dgschwindturo) [reply kotlin.js:11]: Agreed, having something like `SdkmanTool` that extends `Tool` from which `Java` and `Kotlin` would then extend `SdkmanTool` makes good sense, have done this in Java a million times in the past.

**Pull request closed without merging**